### PR TITLE
Add `created` and `updated` fields

### PR DESCRIPTION
**Pull request by rkondratowicz**
### Change description ###
Store information on when draft has be created and last updated.
Using https://en.wikipedia.org/wiki/ISO_8601 format.

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/functionalTest/java/uk/gov/hmcts/reform/draftstore/endpoint/v3/helpers/SampleData.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+import java.time.ZonedDateTime;
+
+public class SampleData {
+    public static Draft draft(String id) {
+        return new Draft(
+            id,
+            "abc",
+            "serviceA",
+            "",
```
Comment (lgonczar) [inline src/functionalTest/java/uk/gov/hmcts/reform/draftstore/endpoint/v3/helpers/SampleData.java]: Can these values normally be empty?

File: src/main/java/uk/gov/hmcts/reform/draftstore/data/DraftStoreDAO.java [hunk 2]
```diff
@@ -47,12 +49,15 @@ public DraftStoreDAO(NamedParameterJdbcTemplate jdbcTemplate) {
...
     public SaveStatus insertOrUpdate(String userId, String service, String type, String newDocument) {
+        ZonedDateTime now = ZonedDateTime.now();
         MapSqlParameterSource params =
             new MapSqlParameterSource()
                 .addValue("userId", userId)
                 .addValue("service", service)
                 .addValue("type", type)
-                .addValue("document", newDocument);
+                .addValue("document", newDocument)
+                .addValue("created", now)
```
Comment (lgonczar) [inline src/main/java/uk/gov/hmcts/reform/draftstore/data/DraftStoreDAO.java]: Can this overwrite `created` on update?

File: src/main/java/uk/gov/hmcts/reform/draftstore/data/DraftStoreDAO.java [hunk 3]
```diff
@@ -156,7 +167,9 @@ public Draft mapRow(ResultSet rs, int rowNumber) throws SQLException {
                 rs.getString("user_id"),
                 rs.getString("service"),
                 rs.getString("document"),
-                rs.getString("document_type")
+                rs.getString("document_type"),
+                rs.getTimestamp("created").toInstant().atZone(ZoneId.systemDefault()),
```
Comment (lgonczar) [inline src/main/java/uk/gov/hmcts/reform/draftstore/data/DraftStoreDAO.java]: We may consider extracting a conversion method. Not that important for now, as there are just two lines like this.

File: src/main/java/uk/gov/hmcts/reform/draftstore/data/DraftStoreDAO.java (hunk 2 above)
Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/draftstore/data/DraftStoreDAO.java]: it's not used in update query

File: src/main/resources/db/migration/V5__add_created_and_updated_columns.sql [hunk 4]
```diff
@@ -0,0 +1,19 @@
+ALTER TABLE draft_document
```
Comment (lgonczar) [inline src/main/resources/db/migration/V5__add_created_and_updated_columns.sql]: We could do these two things (created, updated) in one go, but only if migration time matters here

File: src/functionalTest/java/uk/gov/hmcts/reform/draftstore/data/DataAgent.java [hunk 5]
```diff
@@ -27,8 +27,8 @@ public void setupDocumentForUser(String userId, String type, String document) th
         jsonbObj.setValue(document);
 
         jdbcTemplate.update(
-            "INSERT INTO draft_document (user_id, document_type, service, document) "
-                + "VALUES (:userId, :type, 'cmc', :document)",
+            "INSERT INTO draft_document (user_id, document_type, service, document, created, updated) "
+                + "VALUES (:userId, :type, 'cmc', :document, now(), now())",
```
Comment (timja-kainos) [inline src/functionalTest/java/uk/gov/hmcts/reform/draftstore/data/DataAgent.java]: just putting it out there, what we do in the claim store is always insert at UTC and then use in LocalDateTime which is easier to work with.

Comment (rkondratowicz) [reply src/functionalTest/java/uk/gov/hmcts/reform/draftstore/data/DataAgent.java]: We return timezone info (although it's always UTC) to follow https://hmcts.github.io/restful-api-standards/#126

**Pull request merged**